platform/source/lib/ax/platform/rest.py

```diff
@@ -197,6 +197,9 @@ def cluster_config():
 
 @_app.route("/v1/axmon/cluster/spot_instance_config", methods=['PUT'])
 def put_spot_instance_config():
+    if AXClusterConfig().get_cluster_provider().is_user_cluster():
+        return jsonify({"status": "Spot instances not allowed on user provided K8S clusters."})
```

Comment:
I think you should raise an Exception (AXIllegalOperationException) and the exception handler will deliver the message with a 400 error code. Currently, this will return a 200 code (which is http ok)

Reply:
Good suggestion. Done!